virttest/test_setup.py

```diff
@@ -1301,6 +1300,31 @@ def get_pf_ids(self):
             pf_ids.append(pf_id)
         return pf_ids
 
+    def assign_static_ip(self):
+        """
+        Set the static IP for the PF devices for mlx5_core driver
+        """
+        # This function assigns static IP for the PF devices
+        set_ip = 0
+        pf_devices = self.get_pf_ids()
+        if (self.static_ip):
+            if (not self.start_addr_PF) or (not self.net_mask):
+                raise exceptions.TestSetupFail(
+                    "No IP / netmask found, please populate starting IP address for PF devices in configuration file")
+            ip_addr = netaddr.IPAddress(self.start_addr_PF)
+            set_ip = 1
+        for PF in pf_devices:
+            if set_ip:
+                ifname = utils_misc.get_interface_from_pci_id(PF)
+                ip_assign = "ifconfig %s %s netmask %s up" % (
+                    ifname, ip_addr, self.net_mask)
```

Comment:
so this will try to assign IP even if `self.static_ip` is not configured, also it would probably break here as `ip_addr` is referenced before assignment in case of `self.static_ip` is not configured.

I would suggest to have check of `self.static_ip` and then call `assign_static_ip()` from `sr_iov_setup()`

Reply:
@balamuruhans Addressed the changes suggested, please review. Thanks!